### DISCUSSION TITLE
[WiX4] WixUIExtension support

### DISCUSSION THREAD

**Discussion by jan-tosovsky-cz** [wixtoolset/issues#6516] [Questions] [answered]
I am getting this error when trying to build an installer with a decent UI:
```
wix build -ext WixUIExtension installer.wxs
wix.exe : error WIX0144: The extension 'WixUIExtension' could not be found. Checked paths: WixUIExtension
```
Is this feature available in WiX4-preview.0 or it needs to be activated differently? 

Without that cmdline option there are different errors:
`installer.wxs(50) : error WIX0094: The identifier 'Dialog:PrepareDlg' could not be found. Ensure you have typed the reference correctly and that all the necessary inputs are provided to the linker.`

**Comment by rseanhall** [comment]
The only documentation available on using extensions with wix.exe right now is this design document: https://wixtoolset.org/development/wips/6184-command-line-extension-acquisition-and-cache/

**Comment by jan-tosovsky-cz** [comment, marked as answer]
I finally could add this extension by this command: `wix extension add WixToolset.UI.wixext`
Now it is listed properly:
```
>wix extension list
WixToolset.UI.wixext 4.0.0-preview.0
```
But when building my installer that extension option seems to be ignored:
```
>wix -ext WixToolset.UI.wixext build installer.wxs
wix.exe : error WIX0144: The extension 'WixToolset.UI.wixext' could not be found. Checked paths: WixToolset.UI.wixext
```
However, when specifying the absolute path, it works:
`wix build -ext C:\my-project\.wix\extensions\WixToolset.UI.wixext\4.0.0-preview.0\tools\WixToolset.UI.wixext.dll installer.wxs`

**Reply by rseanhall** [reply]
That's a bug: https://github.com/wixtoolset/issues/issues/6536
